keyboards/skyloong/gk104/q1/q1.c

```diff
@@ -0,0 +1,351 @@
+    {1, SW6_CS13,  SW6_CS14,  SW6_CS15},   //P2
+    {1, SW7_CS13,  SW7_CS14,  SW7_CS15},   //P3
+    {1, SW7_CS16,  SW7_CS17,  SW7_CS18},   //PENT
+
+
+    {1, SW1_CS7,   SW1_CS8,   SW1_CS9},   //Ctrl
+    {1, SW2_CS7,   SW2_CS8,   SW2_CS9},   //Com
+    {1, SW3_CS7,   SW3_CS8,   SW3_CS9},   //Alt
+    {1, SW4_CS7,   SW4_CS8,   SW4_CS9},   //SPAC1
+    {1, SW6_CS7,   SW6_CS8,   SW6_CS9},   //SPAC
+    {1, SW8_CS7,   SW8_CS8,   SW8_CS9},   //SPAC2
+    {1, SW10_CS7,  SW10_CS8,  SW10_CS9},   //Alt
+    {1, SW11_CS7,  SW11_CS8,  SW11_CS9},   //Fn
+    {1, SW12_CS7,  SW12_CS8,  SW12_CS9},   //Menu
+    {1, SW2_CS16,  SW2_CS17,  SW2_CS18},   //Ctrl
+    {1, SW3_CS16,  SW3_CS17,  SW3_CS18},   //Left
+    {1, SW3_CS13,  SW3_CS14,  SW3_CS15},   //Down
+    {1, SW4_CS16,  SW4_CS17,  SW4_CS18},   //Right
+    {1, SW5_CS16,  SW5_CS17,  SW5_CS18},   //P0
+    {1, SW6_CS16,  SW6_CS17,  SW6_CS18}    //PDot
+};
+#endif
+
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+#  ifdef RGB_MATRIX_ENABLE
+    case RGB_TOG:
+         if (record->event.pressed) {
+            switch (rgb_matrix_get_flags()) {
+                    case LED_FLAG_ALL: {
+                        rgb_matrix_set_flags(LED_FLAG_NONE);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    } break;
+                    default: {
+                        rgb_matrix_set_flags(LED_FLAG_ALL);
+                    } break;
+            }
+        }
+        return false;
+    case RGB_VAI:
+       rgb_matrix_set_flags(LED_FLAG_ALL);
+       return true;
+#    endif
+
+    case MO(1):
+      if (record->event.pressed) {
+       FN_ON = 1;
+      } else {
+       FN_ON = 0;
+      }
+      return true;
+
+    case MO(2):
+      if (record->event.pressed) {
+       FN_ON = 1;
+      } else {
+       FN_ON = 0;
+      }
+      return true;
+
+    case MO(3):
+      if (record->event.pressed) {
+       FN_ON = 1;
+      } else {
+       FN_ON = 0;
+      }
+      return true;
+
+    case KC_LGUI:
+      if (FN_ON){
+          if ( record->event.pressed){
+             WIN_LOCK = !WIN_LOCK ; //change win lock state
+            }
+          if (!WIN_LOCK) {
+             IND = IND & (~WINLK_ON);   //Close win lock display
+             s_serial_to_parallel(IND);
+             return false; //windows key locked do nothing
+            }
+          s_serial_to_parallel(IND);
+        }
+      if (WIN_LOCK) {
+             IND = IND | WINLK_ON;   //Open win lock display
+             s_serial_to_parallel(IND);
+             return false; //windows key locked do nothing
+            }
+      return true;  // continue all further processing of this key
+
+    case TO(0):
+       set_single_persistent_default_layer(0);
+       return true;
+
+    case TO(1):
+       set_single_persistent_default_layer(1);
+       return true;
```

Comment:
Should not override default behaviour of core keycodes.

Reply:
You should avoid resolving this convo, as you have not removed the offending code. `TO(x)` should **never** be changed to call `set_single_persistent_default_layer`.